src/workloads/c2c/MongosyncScripts.yml

```diff
@@ -12,88 +12,35 @@ Description: |
         Path: ./MongosyncScripts.yml
         Key: StartMongosync
 
-  Note that the polling scripts start with set -eou pipefail so that any
-  subcommand run during the script failing will cause the entire script to fail.
-  This way if mongosync crashes during the test, it will stop the test execution
-  immediately and genny will bubble up the error and fail the test
-
-  For curl commands we pass -s and -S so that an errors are still reported (-S)
-  but download progress stats are not logged to stderr
-
 StartMongosync:
   Repeat: 1
-  Command: sh
+  Command: python3
   MetricsName: StartMongosync
-  Script: |
-    curl -s -S -X POST http://10.2.0.160:27182/api/v1/start \
-      -H 'Content-Type: application/json' \
-      -d '{"Source": "cluster0", "Destination": "cluster1"}'
+  ScriptPath: ./src/genny/src/workloads/c2c/mongosync_helpers.py -m start
 
 PollForCEA:
   Repeat: 1
-  Command: sh
+  Command: python3
   MetricsName: PollForCEA
-  Script: |
-    set -eou pipefail
-
-    getState() {
-      curl -s -S http://10.2.0.160:27182/api/v1/progress | jq '.progress.info' --raw-output
-    }
-
-    state=$(getState)
-    while [ "$state" != "change event application" ]
-    do
-      echo "Polling for change event application, current state=$state"
-      sleep 1;
-      state=$(getState)
-    done
+  ScriptPath: ./src/genny/src/workloads/c2c/mongosync_helpers.py -m poll_for_cea
 
 DrainWrites:
   Repeat: 1
-  Command: sh
+  Command: python3
   MetricsName: DrainWrites
-  Script: |
-    set -eou pipefail
-
-    getLag () {
-      curl -s -S http://10.2.0.160:27182/api/v1/progress | jq '.progress.lagTimeSeconds'
-    }
-
-    lagTime=$(getLag)
-    while [ $lagTime -gt 0 ];
-    do
-      echo "Waiting for writes to drain, currentLag=$lagTime"
-      sleep 1;
-      lagTime=$(getLag)
-    done
+  ScriptPath: ./src/genny/src/workloads/c2c/mongosync_helpers.py -m drain_writes
 
 Commit:
   Repeat: 1
-  Command: sh
+  Command: python3
   MetricsName: Commit
-  Script: |
-    curl -s -S -X POST http://10.2.0.160:27182/api/v1/commit \
-      -H 'Content-Type: application/json' \
-      -d '{}'
+  ScriptPath: ./src/genny/src/workloads/c2c/mongosync_helpers.py -m commit
 
 WaitForCommit:
   Repeat: 1
-  Command: sh
+  Command: python3
   MetricsName: WaitForCommit
-  Script: |
-    set -eou pipefail
-
-    getState() {
-      curl -s -S http://10.2.0.160:27182/api/v1/progress | jq '.progress.state' --raw-output
-    }
-
-    state=$(getState)
-    while [ $state != "COMMITTED" ]
-    do
-      echo "Waiting for commit to finish, state=$state"
-      sleep 1;
-      state=$(getState)
-    done
+  ScriptPath: ./src/genny/src/workloads/c2c/mongosync_helpers.py -m wait_for_commit
```

Comment:
(nit) Given the args later on, I'm not sure if `ScriptPath` is appropriate for this. Should it just be `ScriptArgs` or so?

Reply:
Alternatively, should it be split into scriptpath and script args?

Reply:
having a way to include things could be interesting - e.g. what if we could have something like `Include: script.py` that has helper functions in it, and then have the python code inline in the yml to invoke the methods how we'd like. That helps avoid having to write an argument parser for every new script

Reply:
Made some updates to this, now there's a folder for python actors and it gets added to genny's PYTHONPATH. The yaml here now just imports the appropriate python actor and calls the function it needs in line.

Reply:
Okay so maybe hot-take, but I think a script Actor would look like this:

```yaml
Actors:
- Type: Script
  Script: src/my-repo/foo-bar.sh # no args!
```

And then the API for the script is:

1. Script is called from work dir (`.`) with the `v1 setup` arg at constructor time. This can do venv or whatever, no need to muck with PYTHONPATH and risk contamination of Genny's python. (The `v1` is there to version this little protocol so the Actor and scripts can evolve over time.)
2. Script is called with `begin` at the start of the workload. It is sent `loop X` to stdin for each iteration with `X` as the phase number.
3. Script outputs metrics data to stdout and log output to stderr.

Here's an example bash script that shows how it could be used:

```bash
#!/usr/bin/env bash

set -eou pipefail

# Called at constructor time
if [[ "$1" == "setup" ]]; then
  setup-venv
  echo "Setup venv" > /dev/stderr
fi

# Called before any Phases start
if [[ "$1" == "begin" ]]; then
  # Cal
  while read -r line; do
    echo "{Metrics: Foo}"
  done
fi

if [[ "$1" == "end" ]]; then
  cleanup-venv
fi
```

Again, there are many other ways to do this and there are obvious limitations with this, but imho it's a nice balance between simplicity and functionality.